### PR TITLE
RAJA Update

### PR DESCRIPTION
**Pull request by pranav-sivaraman**
Make use of Umpire instead of CHAI for memory allocations.
Also add support for other backends such as HIP/SYCL/OpenMP

- [x] HIP Support
- [x] OpenMP Support

SYCL Support is probably not gonna get done since it's not an option in Spack yet.

### CODE REVIEW

File: src/raja/RAJAStream.cpp [hunk 1]
```diff
@@ -9,67 +9,30 @@
...
-  free(d_b);
-  free(d_c);
-#else
-  cudaFree(d_a);
-  cudaFree(d_b);
-  cudaFree(d_c);
-#endif
+    rm.getAllocator(d_a).deallocate(d_a);
+    rm.getAllocator(d_b).deallocate(d_b);
+    rm.getAllocator(d_c).deallocate(d_c);
```
Comment (jhdavis8) [inline src/raja/RAJAStream.cpp:26]: Indentation trouble. Looks like this file uses 2 spaces to indent, these lines are four spaces.

File: src/raja/RAJAStream.hpp [hunk 2]
```diff
@@ -8,65 +8,64 @@
...
 template <class T>
-class RAJAStream : public Stream<T>
-{
+class RAJAStream : public Stream<T> {
   protected:
     // Size of arrays
-  const int array_size;
-  const RangeSegment range;
+	const int array_size;
+	const RAJA::TypedRangeSegment<int> range;
```
Comment (jhdavis8) [inline src/raja/RAJAStream.hpp]: These lines are indented with 8-wide tabs, this file appears to use 4 spaces.

File: src/raja/model.cmake [hunk 3]
```diff
@@ -1,146 +1,38 @@
...
-        set(RAJA_ENABLE_TESTS OFF CACHE BOOL "")
-        set(RAJA_ENABLE_EXAMPLES OFF CACHE BOOL "")
-        set(RAJA_ENABLE_REPRODUCERS OFF CACHE BOOL "")
-        set(RAJA_ENABLE_EXERCISES OFF CACHE BOOL "")
-        set(RAJA_ENABLE_DOCUMENTATION OFF CACHE BOOL "")
-        set(RAJA_ENABLE_BENCHMARKS OFF CACHE BOOL "")
-        set(RAJA_ENABLE_CUDA ${RAJA_ENABLE_CUDA} CACHE BOOL "" FORCE)
-
-        if (ENABLE_CUDA OR RAJA_ENABLE_CUDA)
+    set(CMAKE_CXX_STANDARD 14)
```
Comment (jhdavis8) [inline src/raja/model.cmake:10]: A couple lines here also need indentation fixes. This file uses 4 spaces per level

**Review by jhdavis8 — CHANGES_REQUESTED** [review]

File: src/raja/model.cmake (hunk 3 above)
Comment (pranav-sivaraman) [reply src/raja/model.cmake:10]: It's already 4 spaces per level? I don't see the issue? Maybe the diff shows wrong indentation?

Comment (jhdavis8) [reply src/raja/model.cmake:10]: I'll take a look at what I see in a local checkout, this is weird

Comment (jhdavis8) [reply src/raja/model.cmake:10]: Ok I see what's going on, I just misread the diff, apologies

**Review by jhdavis8 — APPROVED** [review]
LGTM

**Pull request merged**